### ISSUE TITLE
ENH: add support for snake case in pandas.Series.str.title

### ISSUE THREAD

**Issue by jeffolsi** [pandas-dev/pandas#33826]
add support for snake case in `pandas.Series.str.title`

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.title.html

so that :
s = pd.Series(['snake', 'CAPITALS', 'IdNumber'])

will be:
```
s
0                 snake
1              capitals
2              id_number
dtype: object
```

This is very much needed when we want to create database columns and they can't have capital letters. lower() isn't good as well because `nextExecutionDateForYearlyBatch` will be impossible to read : `nextexecutiondateforyearlybatch`

Basically I'm looking to do this:
```
import pandas as pd
import janitor
df = pd.DataFrame(...).clean_names()
```
```
Columns before: First Name, Last Name, Employee Status, Subject
Columns after: first_name, last_name, employee_status, subject
```
https://pyjanitor.readthedocs.io/reference/janitor.functions/janitor.clean_names.html
within the `pandas` domain without the need for `janitor` package.

**Comment by rohitkg98** [comment]
Do you mean snake case? Also, this could be easily tackled by `.apply`, if needed I could create a PR for this functionality but I don't think it should be in the lib.

**Comment by jeffolsi** [comment]
@rohitkg98 yes. Why shouldn't it be? seems like a natural extension of the options already available :
```
Series.str.lower
Converts all characters to lowercase.

Series.str.upper
Converts all characters to uppercase.

Series.str.title
Converts first character of each word to uppercase and remaining to lowercase.

Series.str.capitalize
Converts first character to uppercase and remaining to lowercase.

Series.str.swapcase
Converts uppercase to lowercase and lowercase to uppercase.

Series.str.casefold
Removes all case distinctions in the string.
```

**Comment by mroeschke** [comment]
pandas string functions mirror the standard library, and it doesn't appear that the standard library has string support for snake case.

I agree with @rohitkg98 and think this specific use case can be tackled with custom function.

**Comment by TomAugspurger** [comment]
Agreed with @mroeschke. If this is added to the language then we'll add it in pandas as well. I don't think we would want to implement it on our own though.

**Comment by jeffolsi** [comment]
@TomAugspurger the logic doesn't have to be written in Pandas but can be utilised from another python lib like: https://pypi.org/project/stringcase/

**Comment by TomAugspurger** [comment]
Pandas doesn't depend on that library.

You can, however, apply it's methods to a Series with Series.apply.
